### ISSUE TITLE
Mage talent Hot streak

### ISSUE THREAD

**Issue by Kenniflopper** [gamefreedomgit/Maelstrom#3791]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**
Hot streak is right now really bugged.

**How to reproduce:**
check how many procs u get constantly off scorches on a target dummy / player, i feel like the proc chance is WAY to high, getting procs after procs.

**How it should work:**
there should be some sort of chance for the talent to proc, i feel like mages shit out pyro procs right now and this was an issue on the old atlantis server aswell.

**Database links:**

**Comment by Kenniflopper** [comment]
You can also test this by using frost nova on target dummy and only using fire blast (remember to spec into shatter talent in frost tree aswell)

**Comment by mipanka** [comment]
the talent should have a linear dependence on the crit chance of the mage. More crit chance, less talent proc chance and vice versa.

**Comment by Pornoodemon** [comment]
I don’t know exactly how it works, but here is a comment from a developer from another cataclysm project that has been in the wow bug-fix thread for probably 20 years.
"
Pelf. Developer.

Checked, everything works.
The fire path's proc chance is reduced based on your crit chance. At 40% crit, the chance to proc a fire path is about 4%. At 9 crit, the chance of triggering a fire path is about 54%."

**Comment by mipanka** [comment]
I did test it did 100 crit attack with my 21.86% crit chance was ~43-45% chance prock talents hot streak

**Comment by Pornoodemon** [comment]
Theoretically, you can check otherwise. Take test items on a crit to score 100% crit or close to it. Use only "Hot Streak"
According to the linear dependence, there should be no procs.
If there are procs, then something is wrong

**Comment by Kenniflopper** [comment]
as pornodemon says thats right

**Comment by Pornoodemon** [comment]
> Я проверил, что он нанес 100 критических ударов с моим шансом критического удара 21,86%, что было ~ 43-45% шанса прока талантов.
> 
> ps (-1.7106 * 0.2186 + 0.7893) * 100 где 0.2186 мой шанс крита и брать 41.536284, но без пруфов, пофигу

http://mmoboom.ru/games/archive/4442-mag-ognya-v-cataclysm
He derived the formula from here

**Comment by Pornoodemon** [comment]
![image](https://github.com/gamefreedomgit/Maelstrom/assets/133828891/10f3c8b7-abe6-4a3c-85e2-2f47a4cde3f0)
According to this scheme, there will be no more procs after 50% crit
http://mmoboom.ru/games/archive/4442-mag-ognya-v-cataclysm

**Comment by Kenniflopper** [comment]
@UltraNix any news on this?

**Comment by bonfax4** [comment]
https://web.archive.org/web/20140511230546/http://forums.elitistjerks.com/topic/109377-cataclysm-fire-mage-compendium/

> T3 Hot Streak gives no benefit above ~45% Crit
> T3 and T4 cross over at about ~34% crit chance. That is, below ~34% crit, T3 is more valuable. Above that point T4 is more valuable. This does not mean you should ever skip picking up the T4 talent if your Crit level is below ~34%. Just take note that there's a crossover threshold, where one talent becomes more valuable than the other. They're still both talents you need to pick up regardless.
> Below ~19% crit, T4 HS gives you less than ~1% chance per cast for a HS proc, if you have T3 HS.
> Below ~9% crit, T4 HS gives you less than ~0.1% chance per cast for a HS proc, if you have T3 HS.
> Several diagrams explaining the above points were made by Shaewyn. Click the spoiler tag below to see them. For his full analysis and commentary, visit the link above.

![image](https://github.com/gamefreedomgit/Maelstrom/assets/16911730/97eac96f-f34d-41b0-bde1-60692a89d35a)
![image](https://github.com/gamefreedomgit/Maelstrom/assets/16911730/4c8b6ba6-7b48-4543-939d-c54db4f12a18)

Math they went over [here](http://web.archive.org/web/20120208065212/http://elitistjerks.com/f75/t104767-fire_cataclysm_discussion_op_updated_release/p12/) on the latest posts by Shaewyn.

**Comment by mipanka** [comment]
> https://web.archive.org/web/20140511230546/http://forums.elitistjerks.com/topic/109377-cataclysm-fire-mage-compendium/
> 
> > T3 Hot Streak gives no benefit above ~45% Crit
> > T3 and T4 cross over at about ~34% crit chance. That is, below ~34% crit, T3 is more valuable. Above that point T4 is more valuable. This does not mean you should ever skip picking up the T4 talent if your Crit level is below ~34%. Just take note that there's a crossover threshold, where one talent becomes more valuable than the other. They're still both talents you need to pick up regardless.
> > Below ~19% crit, T4 HS gives you less than ~1% chance per cast for a HS proc, if you have T3 HS.
> > Below ~9% crit, T4 HS gives you less than ~0.1% chance per cast for a HS proc, if you have T3 HS.
> > Several diagrams explaining the above points were made by Shaewyn. Click the spoiler tag below to see them. For his full analysis and commentary, visit the link above.
> 
> ![image](https://user-images.githubusercontent.com/16911730/245640901-97eac96f-f34d-41b0-bde1-60692a89d35a.png) ![image](https://user-images.githubusercontent.com/16911730/245640950-4c8b6ba6-7b48-4543-939d-c54db4f12a18.png)
> 
> Math they went over starting on this page [here](http://web.archive.org/web/20120208065212/http://elitistjerks.com/f75/t104767-fire_cataclysm_discussion_op_updated_release/p12/)

according to the site, at 0.228 crit there should be a chance of about 0.263
according to my last tests on a small number (~100) of crits, I got 0.2186 with my crit chance, the number of Hot streak is about 0.43

**Comment by UltraNix** [comment]
```
Theoretically, you can check otherwise. Take test items on a crit to score 100% crit or close to it. Use only "Hot Streak"
According to the linear dependence, there should be no procs.
```
I think you forgot about Improved Hot Streak. Even if you won't have procs from basic Hot Streak, with 100% crit chance every 2 crits will proc Hot Streak effect.

**Comment by Kenniflopper** [comment]
Yeah but this test should only include the 1 fire blast / scorch and then 3rd, 5th etc

**Comment by UltraNix** [comment]
And what about T12 4p? - increases HS chance by 15%.

**Comment by Kenniflopper** [comment]
ofc that should be taken into account, but im just saying this specific testing should be only the hot streak talent and be tested as said above

**Comment by UltraNix** [comment]
1. test done with 11.23% critical chance. According to the formula it should give ~60% per crit (6.7% per cast)
![WoWScrnShot_061523_224313](https://github.com/gamefreedomgit/Maelstrom/assets/80540499/b5a5d61f-af93-47b2-9750-ba207e52ee59)
![WoWScrnShot_061523_224319](https://github.com/gamefreedomgit/Maelstrom/assets/80540499/76583d0f-84ff-48db-82ba-48daddab789d)
1000 hits, 127 crits and 73 procs of HS.
73/127 ~58%
73/1000 ~7%

2. test done with 31.23% critical chance. According to the formula it should give ~25% per crit (7.8% per cast)
![WoWScrnShot_061523_225908](https://github.com/gamefreedomgit/Maelstrom/assets/80540499/5fb23262-e8e3-4e80-9643-3c865cf1e9cd)
![WoWScrnShot_061523_225914](https://github.com/gamefreedomgit/Maelstrom/assets/80540499/c1e99c7d-122e-4877-a5f8-92d6e47167d5)
1000 hits, 336 crits and 83 procs of HS.
83/336 ~25%.
83/1000 ~ 8.3%

I see no bug here.

**Comment by bonfax4** [comment]
@UltraNix  31.23% should give you @ 9% chance to proc without improved hot streak  and not 25% ?


What is the formula used for t3 now ? If i look on [posts ](http://web.archive.org/web/20120208064232/http://elitistjerks.com/f75/t110326-cataclysm_fire_mage_compendium/p6/) further ahead and simcraft, the formula posted on the guide failed to be updated as usual.  According to comments posted later and simcraft, the updated Hot streak t3 formula should be -2.73c + 0.95.


> And what about T12 4p? - increases HS chance by 15%.
t12 4p should add a 30% chance to Hot streak. 15% to Brain freeze. [Source](https://web.archive.org/web/20110811225714/http://elitistjerks.com/f75/t110326-cataclysm_fire_mage_compendium/p21/#post1949375)

**Comment by bonfax4** [comment]
Yup im getting t3 HS procs at @ 35% char panel crit, which should be impossible if the above formula is correct

**Comment by UltraNix** [comment]
```
@UltraNix 31.23% should give you @ 9% chance per crit to proc without improved hot streak and not 25% ?
```
25% per crit, 9% per cast as I show on screenshots. And it fit the formula from 1. post, as well as the charts.

`
What is the formula used for t3 now ? If i look on [posts ](http://web.archive.org/web/20120208064232/http://elitistjerks.com/f75/t110326-cataclysm_fire_mage_compendium/p6/)further ahead and simcraft, the formula posted on the guide failed to be updated as usual. According to comments posted later and simcraft, the updated Hot streak t3 formula should be -2.73c + 0.95.
`
The formula from the 1.post is used.
`
(-1.7106 * crit + 0.7893) * 100.
`

**Comment by UltraNix** [comment]
@bonfax4 You are testing without improved HS?

**Comment by bonfax4** [comment]
Yes

**Comment by UltraNix** [comment]
35% crit chance gives you 6% chance per cast to proc, according to the formula

**Comment by UltraNix** [comment]
`T3 Hot Streak gives no benefit above ~45% Crit`

**Comment by UltraNix** [comment]
```
And what about T12 4p? - increases HS chance by 15%.

t12 4p should add a 30% chance to Hot streak. 15% to Brain freeze. [Source](https://web.archive.org/web/20110811225714/http://elitistjerks.com/f75/t110326-cataclysm_fire_mage_compendium/p21/#post1949375)
```
Yes - it adds 30% and that part is okay. My bad.

**Comment by bonfax4** [comment]
Yeah that 45% stuff is outdated, was based on the math done during early prepatch cata. If the new formula is correct, it falls off around 34.7% crit.

**Comment by UltraNix** [comment]
What is the most common crit chance on our server for fire mage?

**Comment by bonfax4** [comment]
Asked a pvp fire mage and he had 24% with molten armor

https://gyazo.com/5dce13e60ab56442e4209a213013826a

**Comment by bonfax4** [comment]
That gap will increase with higher crit % in pve scenarios i'd guess.

**Comment by UltraNix** [comment]
I may change that formula, but you won't see the difference.

**Comment by bonfax4** [comment]
To note that the person that's discussing the new formula on the post i linked ( Shaewyn ) is also the same person that posted and assisted in testing the 1st initial formula.

**Comment by bonfax4** [comment]
Yeah only around the 20%-35% mark you'd see a bigger difference between the old and new formula formula but nothing more than 4% increase. If nothing else is broken numbers won't differ much. 

Old
![procs_vs_cr](https://github.com/gamefreedomgit/Maelstrom/assets/16911730/b89528ba-0247-44f3-ba4a-44beb7f6c3e7)

New
![hs_ppcr_ppc-1](https://github.com/gamefreedomgit/Maelstrom/assets/16911730/faa99994-2339-45a7-b8df-dc8eb7fe5ff1)

**Comment by bonfax4** [comment]
One thing that prob still needs testing is if t3 procs reset t4 procs counter.

**Comment by Kenniflopper** [comment]
@UltraNix news on this

**Comment by UltraNix** [comment]
What news? I already made tests and see nothing wrong (as you can see above).

**Comment by UltraNix** [comment]
No further info - closing.
